hydra_python_core/doc_writer.py

```diff
@@ -629,6 +632,46 @@ def generate(self) -> Dict[str, Any]:
         return error
 
 
+class HydraLink():
+    """Template for a link property."""
+
+    def __init__(
```

Comment:
We don't need all these fields. Hydra:Link is defined in the spec, so we don't have to add it separately to the API Doc. There is no need for all these fields in a link

Reply:
I think we have a misunderstanding here, HydraLink creates instances of predicates with type `hydra:Link`, pretty much similar to what we are already doing with HydraClass, Collection, etc.

Reply:
What I mean to say is that we don't really need to define hydra:Link in the API documentation. Doc writer does not create instances of the class/resource, it only adds an entry for the class in the API Documentation. The `id_` field in this case is not an instance UID, it is the URI of the class definition. For hydra:Link, we don't define it in the API Documentation. It is already defined in the spec. We can use it directly. The only reason we define a HydraLink class is for us to be able to generate instances(here instances are links to existing resource, not an entry in the API documentation which is the case with HydraClass). Does this clear it up?

Reply:
>What I mean to say is that we don't really need to define hydra:Link in the API documentation. Doc writer does not create instances of the class/resource, it only adds an entry for the class in the API Documentation.

Sorry for the poor wording, similar to class, the hydraLink also adds an entry of a predicate with type `hydra:Link` to the ApiDoc.

>For hydra:Link, we don't define it in the API Documentation. It is already defined in the spec. We can use it directly. The only reason we define a HydraLink class is for us to be able to generate instances(here instances are links to existing resource, not an entry in the API documentation which is the case with HydraClass). Does this clear it up?

We need to give access to the api designer to create/add hydra:Link properties in the APIDOC. Similar to class he can create a predicate with `hydra:Link` and utilize it as a `property` of SupportedProperty in any class's `supportedProperty`.